pennylane/_grad.py

```diff
@@ -288,9 +367,60 @@ def circuit(x, y):
 
     As expected, the tuple was unpacked and we directly received the Jacobian of the
     QNode with respect to ``y``.
+
+    We can also compute the Jacobian transformation inside a :func:`~.qjit` decorated program:
+
+    .. code-block::
+
+        dev = qml.device("lightning.qubit", wires=1)
+
+        @qml.qjit
+        def workflow(x):
+            @qml.qnode(dev)
+            def circuit(x):
+                qml.RX(np.pi * x[0], wires=0)
+                qml.RY(x[1], wires=0)
+                return qml.probs()
+
+            g = qml.jacobian(circuit)
+            return g(x)
+
+    >>> workflow(np.array([2.0, 1.0]))
+    array([[-1.32116540e-07,  1.33781874e-07],
+           [-4.20735506e-01,  4.20735506e-01]])
+
+    You can further compute the Jacobian transformation using other supported differentiation
+    methods by :func:`catalyst.jacobian`.
+
+    .. code-block::
+
+        @qml.qjit
+        def workflow(x):
+            @qml.qnode(dev)
+            def circuit(x):
+                qml.RX(np.pi * x[0], wires=0)
+                qml.RY(x[1], wires=0)
+                return qml.probs()
+
+            g = qml.jacobian(circuit, method="fd", step_size=0.3)
+            return g(x)
+
+    >>> qml.qjit(workflow)(np.array([2.0, 1.0]))
+    array([[-0.37120096, -0.45467246],
+            [0.37120096,  0.45467246]])
     """
     # pylint: disable=no-value-for-parameter
 
+    if active_jit := compiler.active_compiler():
+        available_eps = compiler.AvailableCompilers.names_entrypoints
+        ops_loader = available_eps[active_jit]["ops"].load()
+        return ops_loader.jacobian(func, method=method, h=step_size, argnum=argnum)
```

Comment:
it seems like `jacobian` is part of a PL compiler API. is this API documented somewhere? If so, it should be mentioned in the "Adding a compiler" section of the compiler module docs below. If not, I think it would be a good thing to add

Reply:
We barely discussed `ops` entry_points and generally `pennylane.compilers` specs as they're still some ongoing discussions around the addition of new compilers to PL. I'm not worried about this API for now, as it may be re-designed within few months.

Reply:
in that case, I am concerned about the existence of an "Adding a compiler" section in the docs altogether. We don't seem to be truly ready for a second PL compiler implementation, because we haven't really defined an interface of any kind. I think this totally makes sense! There's no reason to define an API the first time you do something. But the second time around, some interface standardization would go a long way. If some third-party org wanted to build the second-ever PL compiler, I think they'd have a really hard time integrating it with PennyLane core. Are we sure we want to advertise it in PennyLane docs at all?

Reply:
I'd like to keep this section as is - but it's a fair worry.

Let's just add a short warning box here saying 'this API is experimental and subject to change'.

Reply:
I think that's fine. once all PRs are merged to master, I'd like to catalogue the functions that are integrated into the PL frontend and formally list them (categorized by entrypoint, of course) along with their input/output types to make things easier for compiler devs. even if it's subject to change, at least having a central list of what could/should be implemented would be nice.

for example, if you (a 3P compiler developer) don't have `jacobian` implemented, the user will get a very cryptic error. given how experimental that is, I'd say fair enough. BUT if you (the compiler dev) feel so inspired, you can at least implement the (experimental) API and just raise nicer errors saying "hey we haven't implemented jacobian yet". or we (PL devs) make the API more formal and do that error-handling ourselves, but I wouldn't worry about implementing that in such an experimental phase

Reply:
@timmysilv as we discussed offline in the morning, let's just add "experimental and subject to change..." at the beginning of this section until we have a better understanding of the API. 

I'm also not worried about the error messages, as it would throw `AttributeError` if for example the  `jacobian` method is not implemented in catalyst:

``` python
E           AttributeError: module 'catalyst' has no attribute 'jacobian'.
```